### PR TITLE
Update libosmium to 2.18.0. Fix problem with Docker image build.

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
# Issue

closes https://github.com/Project-OSRM/osrm-backend/issues/6302

## Tasklist

 - [ ] CHANGELOG.md entry ([How to write a changelog entry](http://keepachangelog.com/en/1.0.0/#how))
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [ ] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md))
 - [ ] review
 - [ ] adjust for comments
 - [ ] cherry pick to release branch

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: .github/workflows/osrm-backend.yml [hunk 1]
```diff
@@ -84,7 +84,7 @@ jobs:
             runs-on: ubuntu-20.04
             BUILD_TOOLS: ON
             BUILD_TYPE: Debug
-            CLANG_VERSION: 5.0.0
```
Comment (SiarheiFedartsou) [inline .github/workflows/osrm-backend.yml:87]: Had build errors on 5.0, I think it is okay to update it to 6.0.

File: .github/workflows/osrm-backend.yml [hunk 2]
```diff
@@ -47,6 +47,16 @@ jobs:
         ./scripts/format.sh && ./scripts/error_on_dirty.sh
         node ./scripts/validate_changelog.js
         npm run docs && ./scripts/error_on_dirty.sh
+  
+  docker-image:
```
Comment (SiarheiFedartsou) [inline .github/workflows/osrm-backend.yml:51]: Checks that we won’t have problem like this again

Comment (mjjbell) [reply .github/workflows/osrm-backend.yml:51]: Let's add `continue-on-error: false` to this job in that case.

Comment (SiarheiFedartsou) [reply .github/workflows/osrm-backend.yml:51]: Done.

**Review by mjjbell — APPROVED** [review]

**Pull request merged**